### PR TITLE
bpo-38659: [Enum] add _simple_enum decorator

### PR DESCRIPTION
**Pull request by ethanfurman**
add:

- `_simple_enum` decorator to transform a normal class into an enum
- `_test_simple_enum` function to compare
- `_old_convert_` to enable checking `_convert_` generated enums

`_simple_enum` takes a normal class and converts it into an enum:

    @simple_enum(Enum)
    class Color:
        RED = 1
        GREEN = 2
        BLUE = 3

`_old_convert_` works much like `_convert_` does, using the original logic:

    # in a test file
    import socket, enum
    CheckedAddressFamily = enum._old_convert_(
            enum.IntEnum, 'AddressFamily', 'socket',
            lambda C: C.isupper() and C.startswith('AF_'),
            source=_socket,
            )

`test_simple_enum` takes a traditional enum and a simple enum and
compares the two:

    # in the REPL or the same module as Color
    class CheckedColor(Enum):
        RED = 1
        GREEN = 2
        BLUE = 3

    _test_simple_enum(CheckedColor, Color)

    _test_simple_enum(CheckedAddressFamily, socket.AddressFamily)

Any important differences will raise a `TypeError`

https://bugs.python.org/issue38659

### CODE REVIEW

File: Doc/library/enum.rst [hunk 1]
```diff
@@ -117,6 +117,19 @@ Module Contents
 
       Enum class decorator that ensures only one name is bound to any one value.
 
+   :func:`simple_enum`
```
Comment (gpshead) [inline Doc/library/enum.rst]: I suggest we not make this a public API for at least the first release we see if we want to live with this approach.  So I wouldn't document it yet and would add an _ prefix to both names.  That way if we wind up doing something else later it is easy to undo without a drawn out deprecation.

File: Doc/library/enum.rst [hunk 2]
```diff
@@ -622,3 +635,29 @@ Utilites and Decorators
       ...
       ValueError: duplicate values found in <enum 'Mistake'>: FOUR -> THREE
 
+.. decorator:: simple_enum
+
+   A :keyword:`class` decorator that converts a normal class into an enum::
+
+      >>> from enum import Enum, simple_enum
+      >>> @simple_enum(Enum):
```
Comment (gpshead) [inline Doc/library/enum.rst]: if we were keeping the docs... extra trailing : and I believe auto() needs to be imported below?

File: Lib/enum.py [hunk 3]
```diff
@@ -390,13 +391,15 @@ def __prepare__(metacls, cls, bases, **kwds):
                     )
         return enum_dict
 
-    def __new__(metacls, cls, bases, classdict, boundary=None, **kwds):
+    def __new__(metacls, cls, bases, classdict, boundary=None, _simple=False, **kwds):
```
Comment (gpshead) [inline Lib/enum.py]: make `_simple` a keyword only arg.

**Review by gpshead — COMMENTED** [review]
neat approach.  i haven't done a thorough review.  just high level skimming.  we should loop in enum experts.

File: Doc/library/enum.rst (hunk 1 above)
Comment (ethanfurman) [reply Doc/library/enum.rst]: Sounds good.

File: Lib/enum.py (hunk 3 above)
Comment (ethanfurman) [reply Lib/enum.py]: Good idea.

**Comment by ethanfurman** [comment]
> neat approach. i haven't done a thorough review. just high level skimming. we should loop in enum experts.

I am the enum expert.  :-)

**Comment by ethanfurman** [comment]
Not sure what happened -- was only supposed to change 20 files, not 234.  Reverting.

**Comment by gvanrossum** [comment]
Umm... arid something go wrong here? 109 commits?!

**Comment by ethanfurman** [comment]
@methane - so instead of `git rebase master` do `git merge master` ?  Thank you.

**Pull request merged**